examples/chip-tool/commands/common/Command.cpp

```diff
@@ -54,6 +58,59 @@ bool Command::InitArguments(int argc, char ** argv)
     return isValidCommand;
 }
 
+bool Command::InitOptionalArguments(int & argc, char ** argv)
+{
+    bool isValidCommand = false;
+    int optArgsCount    = static_cast<int>(mOptionalArgs.size());
+    int argsCount       = static_cast<int>(mArgs.size());
+    int optionalArgC    = argc - argsCount;
+    int expectedArgs    = optionalArgC;
+    int optInits        = 0;
+
+    // If no optional args passed in, in return true.
+    if (optionalArgC == NO_OPTIONAL_ARGS)
+        return true;
+
+    // If too many optinoal args passed in, return false.
+    if (optArgsCount * OPTIONAL_ARGS_OFFSET < optionalArgC)
+    {
+        ChipLogError(chipTool, "Too Many arguments passed in.");
+        return false;
+    }
+    // Iterate the total configured optional args
+    for (int i = 0; i < optArgsCount; i++)
+    {
+        Argument arg = mOptionalArgs.at((size_t) i);
+        // Check to see if the argument matches what has been configured.
+        if (argv[optInits * OPTIONAL_ARGS_OFFSET + argsCount] != nullptr &&
+            strcmp(arg.name, argv[optInits * OPTIONAL_ARGS_OFFSET + argsCount]) == 0)
```

Comment:
What's keeping this from indexing out of bounds into argv?  I don't see anything preventing that.  Most simply, if `optionalArgC` == 1 but we have two possible optional args, once we find the first one the second one will index out of bounds.

The `optionalArgC == NO_OPTIONAL_ARGS` works around this for the special case of no optional args passed in at all, which is why this might not have been caught....  We should probably take out that special-case and have better testing for "we have exhausted all the args", and that testing needs to be inside this loop.